### PR TITLE
SNOW-830394 adding better configuration file changelog entry

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-830394

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Just a better explanation to what the new feature can do

### CODE REVIEW

File: DESCRIPTION.md [hunk 1]
```diff
@@ -10,6 +10,21 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 - v3.0.5(TBD)
   - Added the ability to read Snowflake's central configuration file.
+    - The configuration file (`config.toml`) location gets resolved the following way:
+      - If the folder defined by the environmental variable `SNOWFLAKE_HOME` (defaults to `~/.snowflake`) exists then we look for the configuration file in there.
+      - We fall back to `platformdirs` defaults. Which means:
```
Comment (sfc-gh-stan) [inline DESCRIPTION.md:15]: Otherwise, we fall back?

**Review by sfc-gh-stan — APPROVED** [review]
One NIT, otherwise LGTM.

File: DESCRIPTION.md [hunk 2]
```diff
@@ -10,6 +10,21 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
+        - On Windows: `%USERPROFILE%\AppData\Local\snowflake\`
+      - Exact resolution can always be checked by running `python -c "from snowflake.connector.constants import CONFIG_FILE; print(str(CONFIG_FILE))"`
+    - The Python connector supports loading connection parameters from configuration file. An example config file would be:
+      ```toml
+      [connections.prod]
+      account = "my_account"
+      user = "my_user"
+      password = "my_password"
+      ```
+      Then the connection name (`prod` in this case) can be provided to `connect` with the new `connection_name` argument.
```
Comment (sfc-gh-sfan) [inline DESCRIPTION.md:27]: Github renders this in a terrible way :/

**Review by sfc-gh-sfan — APPROVED** [review]

File: DESCRIPTION.md (hunk 2 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:27]: The Markdown file should render correctly, just not the diff 😅

File: DESCRIPTION.md [hunk 3]
```diff
@@ -10,6 +10,21 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 - v3.0.5(TBD)
   - Added the ability to read Snowflake's central configuration file.
```
[the remark this answers is not in the document]
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md]: I've incorporated your comment 👍

File: DESCRIPTION.md [hunk 4]
```diff
@@ -9,7 +9,25 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 # Release Notes
 
 - v3.0.5(TBD)
-  - Added the ability to read Snowflake's central configuration file.
+  - Added a feature that lets you add connection definitions to the `config.toml` configuration file. A connection definition refers to a collection of connection parameters. The connection configuration name must begin with **connections**, similar to the following that defines the parameters for the `prod` connection:
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:12]: It looks like we need to fix the indentation so the new content looks like"

- Added a feature that...
  
    code example

    By default...

      - On Linux
      - On Mac

    You can ...

     code example

- Bumped cryptography...
- Improved OCSP...

**Review by sfc-gh-mheavin — APPROVED** [review]

File: DESCRIPTION.md (hunk 4 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:12]: Fixed the indentation, great catch! 👍

**Pull request merged**